Makefile

```diff
@@ -62,13 +131,22 @@ all: $(PROGRAMS) $(LIBRARIES)
 
 DEPDIR = .deps
 df = $(DEPDIR)/$(*D)/$(*F)
+
+ifeq ($(PE_ENV), GNU)
 %.o: %.cpp
 	@$(MKDIR_P) $(DEPDIR)/$(*D)
 	$(CPU_COMPILE) -MD -o $@ $<
 	@cp $*.d $(df).P; \
 	sed -e 's/#.*//' -e 's/^[^:]*: *//' -e 's/ *\\$$//' \
 		-e '/^$$/ d' -e 's/$$/ :/' < $*.d >> $(df).P; \
 	rm -f $*.d
+else
+%.o: %.cpp
+	@$(MKDIR_P) $(DEPDIR)/$(*D)
+	$(CPU_COMPILE) -o $@ $<
```

Comment:
For the Cray compiler we will have to write manually all the dependencies. The above rule is not enough, because it ignores the header files.